### PR TITLE
Api integration

### PR DESCRIPTION
**Pull request by flacoman91**
depends on https://github.com/cfpb/ccdb5-api/pull/99

### CODE REVIEW

File: src/actions/complaints.jsx [hunk 1]
```diff
@@ -1,18 +1,106 @@
...
+    const viewMode = state.query.tab
+    switch ( viewMode ) {
+      case MODE_MAP:
+        dispatch( getStates() )
+        break
+      // case 'Trends':
+      //   dispatch( getTrends() )
+      //   break
+      case MODE_LIST:
+        dispatch( getComplaints() )
```
Comment (JeffreyMFarley) [inline src/actions/complaints.jsx]: The way this reads, `getComplaints()` is called twice for `MODE_LIST` when `sendQuery()` is called

File: src/actions/__tests__/complaints.spec.jsx [hunk 2]
```diff
@@ -1,12 +1,87 @@
...
+              then: (x) => {
+                onSuccess = (data) => x(data)
+                return {
+                  catch: (y) => {onFail = y}
+                }
+              }
+            }
+          }
+        }
+      })
```
Comment (JeffreyMFarley) [inline src/actions/__tests__/complaints.spec.jsx]: 🙌

File: src/middleware/__tests__/queryManager.spec.jsx [hunk 3]
```diff
@@ -3,69 +3,152 @@ import thunk from 'redux-thunk'
...
-    it('does not query if an action has no metadata', () => {
-      const action = {
-        type: 'FakeAction'
-      }
-      const expectedActions = [ { type: 'FakeAction' } ]
-
-      store.dispatch(action)
-      expect(store.getActions()).toEqual(expectedActions)
-    })
+    describe( 'Unknown Mode', () => {
```
Comment (JeffreyMFarley) [inline src/middleware/__tests__/queryManager.spec.jsx]: 👍

File: src/middleware/__tests__/queryManager.spec.jsx [hunk 4]
```diff
@@ -3,69 +3,152 @@ import thunk from 'redux-thunk'
...
+          const action = {
+            type: 'FakeAction',
+            requery: REQUERY_ALWAYS
+          }
+          const expectedActions = [
+            { type: 'FakeAction', requery: REQUERY_ALWAYS },
+            // put this back when we have the api
+            // { type: 'AGGREGATIONS_API_CALLED', url: "@@API?foo" },
+            { type: 'COMPLAINTS_API_CALLED', url: "@@API?foo" },
+            { type: 'COMPLAINTS_API_CALLED', url: "@@API?foo" }
```
Comment (JeffreyMFarley) [inline src/middleware/__tests__/queryManager.spec.jsx]: At least your test expects both calls

File: src/actions/complaints.jsx (hunk 1 above)
Comment (flacoman91) [reply src/actions/complaints.jsx]: Yes, sorry, that's intentional.  I put getComplaints in there as a placeholder for the aggregations for now and didn't want to have to change it again when the api endpoint is available.

**Pull request merged**